src/cargo/ops/cargo_new.rs

```diff
@@ -514,7 +515,12 @@ fn get_environment_variable(variables: &[&str] ) -> Option<String>{
 }
 
 fn discover_author() -> CargoResult<(String, Option<String>)> {
-    let git_config = GitConfig::open_default().ok();
+    let cwd = env::current_dir()?;
+    let git_config = if let Ok(repo) = GitRepository::discover(&cwd) {
+        repo.config().ok()
```

Comment:
Oh I just means that if `repo.config().ok()` is none, we fall back to `open_default`. So we'd fall back either if `discover` failed or if `config` failed on the opened repository